crates/bevy_ecs/src/world/entity_access/filtered.rs

```diff
@@ -648,6 +684,38 @@ impl<'w, 's> FilteredEntityMut<'w, 's> {
             .flatten()
     }
 
+    /// Gets the "target" entity of this entity via the [`Relationship`] component with the given [`ComponentId`].
+    ///
+    /// **You should prefer to use the typed API where possible and only
+    /// use this in cases where the actual component types are not known at
+    /// compile time.**
+    ///
+    /// [`Relationship`]: crate::relationship::Relationship
+    pub fn get_relationship_by_id(&self, relationship_id: ComponentId) -> Option<Entity> {
+        self.as_readonly().get_relationship_by_id(relationship_id)
+    }
+
+    /// Gets an iterator to the "related" entities of this entity via the [`RelationshipTarget`] component with the given [`ComponentId`].
+    ///
+    /// **You should prefer to use the typed API where possible and only
+    /// use this in cases where the actual component types are not known at
+    /// compile time.**
+    ///
+    /// [`RelationshipTarget`]: crate::relationship::RelationshipTarget
+    pub fn get_relationship_targets_by_id(
+        &self,
+        relationship_target_id: ComponentId,
+    ) -> Option<impl Iterator<Item = Entity> + use<'w>> {
+        self.access
+            .has_component_read(relationship_target_id)
+            // SAFETY: We have read access
+            .then(|| unsafe {
+                self.entity
+                    .get_relationship_targets_by_id(relationship_target_id)
+            })
+            .flatten()
```

Comment:
Why can't this be `self.as_readonly().get_relationship_targets_by_id(relationship_target_id)`, like `get_relationship_by_id` is?  ... Oh, I see, that would borrow from a temporary.  Wait, then how do we do that elsewhere?  ...  Hmm, I think some of the lifetimes are wrong here.  

The mutable methods need to return things that borrow from `'_` instead of `'w`, since otherwise you could do

```rust
let borrows_r1 = filtered_entity.get_relationship_targets_by_id(r1);
let borrows_r1_mutably = filtered_entity.get_mut_by_id(r1);
// Both variables are alive here, because the lifetime in `borrows_r1`
// wasn't tied to the borrow of `filtered_entity`!
```

If we want mutable methods that return `'w`, they need to consume `self`, like `into_mut()` does.  

So, I think this should be:

```suggestion
    ) -> Option<impl Iterator<Item = Entity> + use<'_>> {
        self.as_readonly().get_relationship_targets_by_id(relationship_target_id)
```

Reply:
Thank you! Great catch